boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts

```diff
@@ -203,6 +204,79 @@ i2s1: &flexcomm3 {
 	dma-names = "tx";
 };
 
+/* PCA9420 PMIC */
+&pmic_i2c {
+	status = "okay";
+	compatible = "nxp,lpc-i2c";
+	label = "PMIC_I2C";
+	clock-frequency = <I2C_BITRATE_FAST>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	pca9420: pca9420@61 {
+		label = "PCA9420_PMIC";
+		reg = <0x61>;
+
+		pca9420_sw1: sw1_buck {
+			label = "SW1_BUCK_REG";
+			compatible = "regulator-i2c";
```

Comment:
I've never seen things done this way for an I2C device. Do you have any prior examples elsewhere, e.g. in a Linux binding? I was expecting to see a binding for the PCA9420 itself somewhere.

Reply:
Here's how it is done in the Linux Kernel: https://github.com/torvalds/linux/blob/master/arch/arm64/boot/dts/freescale/imx8mn-evk.dts#L33. For simplicity, I chose to have the DTS declare all PMIC registers, so that this driver could be made generic in the future.

Reply:
Thank you for the pointer! That indeed is more like what I was expecting -- there is a `compatible = "nxp,pca9450b";` in the Linux DTS, for instance.

DTS interoperability with Linux is a goal of Zephyr's (https://docs.zephyrproject.org/latest/guides/dts/design.html#source-compatibility-with-other-operating-systems), so a divergence here needs justification (https://docs.zephyrproject.org/latest/guides/dts/bindings.html#rules-for-mainline-bindings).

I'm not sure I follow the point about simplicity and making things generic, could you please clarify? I'm not too familiar with PMICs in general so sorry in advance for what might be a basic question.